### DISCUSSION TITLE
Does dummy grain create again and again after migrate until silos cache directory expire?

### DISCUSSION THREAD

**Discussion by 0990** [dotnet/orleans#8644] [Q&A]
When grainA  migrate  success from silo1 to silo2, However, silos cache directory lookups and those caches aren't immediately invalidated when a grain is deactivated.
The request to grainA may still forward to silo1 before cached expired.

When silo1 received grainA message：
1，It create a dummy grain，RegisterMesssageTarget,and start activating.
2，In activating process,GrainLocator.Register,found grainA have alreay register in silo2
3，Stop activating, set status Invalid, UnRegisterMessageTarget
4，Forward message to silo2

After UnRegisterMessageTarget, the new grainA message will cause silo1 create dummy grain A again,So It will create again and again util silos cache directory expire?
Was I miss something?

**Comment by gfoidl** [comment]
Is continued in https://github.com/dotnet/orleans/issues/8645
